### PR TITLE
Use file based routing

### PR DESCRIPTION
**Pull request by kenneth-duverge**
### Links
File based routing NextJS docs - https://nextjs.org/docs/app/building-your-application/routing/defining-routes

### What
Switching over to using file based routing that way we can follow NextJS routing conventions.

**Current folder structure:**

```sh
app/
  blog/
     page.tsx <-- /blog route
  list/
  calendar/
components/

```

**Notes:**
- ~We may not want to server render the list page because that happens at build time and we lost the realtime update when creating a new event or deleting one.~ Updated the list page to render on the client since `convex` doesn't have stable support for NextJS server side rendering. There is this [convex article](https://docs.convex.dev/client/react/nextjs/server-rendering) showing how to use convex with react server components but we can revisit this down the line.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,7 +3,9 @@
   "version": "0.1.0",
   "private": true,
   "scripts": {
-    "dev": "next dev",
+    "dev": "npm-run-all --parallel dev:server dev:client",
+    "dev:server": "convex dev",
+    "dev:client": "next dev",
```
Comment (kenneth-duverge) [inline package.json:8]: Now all we need to do is run `bun dev` locally and the next server and convex server will start up in parallel. Prior to this change we had to run `bun dev` in one terminal and `bunx convex dev` in another.

File: app/page.tsx [hunk 2]
```diff
@@ -1,9 +1,3 @@
-import { Newsletter } from '@/components/newsletter';
-
 export default function Home() {
-  return (
-    <>
-      <Newsletter />
-    </>
-  );
+  return <div>Default page</div>;
```
Comment (kenneth-duverge) [inline app/page.tsx:2]: Not sure what we're going to do here yet but we'll leave it as is for now

File: components/view-picker.tsx [hunk 3]
```diff
@@ -1,51 +1,48 @@
...
+}
+
 const LinkButton = ({
   children,
-  to,
-  className,
-}: React.PropsWithChildren<{ className: string; to: 'calendar' | 'list' | 'blog' }>) => {
+  href,
+  className = '',
+  active,
```
Comment (kenneth-duverge) [inline components/view-picker.tsx:19]: I added an `active` prop to handle styling the links when the link is active inside this component.

```
active ? 'bg-accent text-accent-foreground' : ''
```

File: components/view-picker.tsx [hunk 4]
```diff
@@ -1,51 +1,48 @@
...
+      className={`${className} ${active ? 'bg-accent text-accent-foreground' : ''}`}
+    >
+      <Link href={href}>{children}</Link>
     </Button>
   );
 };
 
 export const ViewPicker = () => {
-  const params = useSearchParams();
+  const pathname = usePathname();
```
Comment (kenneth-duverge) [inline components/view-picker.tsx:34]: Docs on `usePathname` https://nextjs.org/docs/app/api-reference/functions/use-pathname#parameters

File: components/view-picker.tsx (hunk 3 above)
Comment (kenneth-duverge) [reply components/view-picker.tsx:19]: Also renamed `to` to `href` to match the `href` prop of the `Link` component.

Comment (daouda-sakh0) [reply components/view-picker.tsx:19]: nice job on the refactor here

File: package.json (hunk 1 above)
Comment (kenneth-duverge) [reply package.json:8]: ### Before

dev-before

### After
dev-after

**Review by daouda-sakh0 — APPROVED** [review]
Reviewed with @kenneth-duverge and giving it a 👍🏿

**Pull request merged**